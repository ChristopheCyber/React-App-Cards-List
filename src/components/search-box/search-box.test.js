import React from 'react';
// import { render } from '@testing-library/react';
import { SearchBox } from './search-box.component.jsx';

//ENZYME fncts: ( we don't use MOUNT and RENDER here )
import { shallow } from "enzyme";


// console.log("mockMethodHandleChange() triggered in search-box.test.js");

describe('Group Script search-box.test.I: ', () => {
    it('Test search-box.test.I.1; Snapshot testing:', () => {
        //mocking methodHandleChange fn :
        const mockMethodHandleChange = jest.fn().mockReturnValue(null);
        expect.assertions(1);
        // const methodHandleChange = mockMethodHandleChange;
        // testing if shallow rendering correctly:
        expect(shallow(<SearchBox propPlaceHolder="Type text for Searching" handleChangeFct={mockMethodHandleChange} />)).toMatchSnapshot();
    });
});
describe('Group Script search-box.test.II: ', () => {
    it('Test search-box.test.II.1; testing Event onChange in <input/> button:', async () => {
        //mocking methodHandleChange fn :
        const mockMethodHandleChange = jest.fn().mockReturnValue(null);
        expect.assertions(1);
        const wrapper = shallow(<SearchBox propPlaceHolder="Type text for Searching" handleChangeFct={mockMethodHandleChange} />);
        // testing :
        //.find() and simulate(events) methods comming with ENZYME:
        wrapper.find('[id="searchBoxId"]').simulate('change');
        await expect(mockMethodHandleChange.mock.calls.length).toBe(1);
    });
});
/*
describe('Group Script search-box.test.II: ', () => {
  it('Test search-box.test.II.1; State testing:', () => {
    expect.assertions(1);
    const wrapper = shallow( <SearchBox propPlaceHolder="Type text for Searching"  mockMethodHandleChange /> );
    // testing :
    //.find() and simulate(events) methods comming with ENZYME:
    // wrapper.find('[id="searchBoxId"]').simulate('click');
    // wrapper.find('[id="searchBoxId"]').simulate('change');
    expect(wrapper.state('picContentType')).toEqual('img-class');
  });
*/

