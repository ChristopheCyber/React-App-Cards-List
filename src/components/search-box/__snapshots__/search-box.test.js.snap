// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group Script search-box.test.I:  Test search-box.test.I.1; Snapshot testing: 1`] = `
<input
  className="search-box"
  id="searchBoxId"
  name="search1"
  onChange={[MockFunction]}
  placeholder="Type text for Searching"
  type="search"
/>
`;
